src/main/java/com/yahoo/sketches/counting/PositiveCountersMap.java

```diff
@@ -0,0 +1,146 @@
+package com.yahoo.sketches.counting;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map.Entry;
+import java.util.ArrayList;
+
+/**
+ * This is a utility class that implements (and abstracts) a set of positive counters. 
+ * The mapping is  Long key -> Long count.
+ * The default value of any key is zero and no negative counters are allowed. 
+ * Non-positive mappings are deleted. 
+ * 
+ * It also support incrementing individual counters and decrementing all counters simultaneously.
+ * This is a convenient and efficient modification intended to be used in FrequentDirection sketching.
+ * 
+ * @author edo 
+ */
+public class PositiveCountersMap{
+  
+  private HashMap<Long,Long> counters;
+  ArrayList<Long> keysToRemove;
+  private long offset;
+  private long nnz;
+
+  /**
+   * Creates empty mappings and default offset = 0.
+   */
+  public PositiveCountersMap(){
+    counters = new HashMap<Long,Long>();
+    keysToRemove = new ArrayList<Long>();
+    offset = 0L;
+  }
+  
+  /**
+   * @return the number of positive counters
+   */
+  public long nnz(){
+    return nnz;
+  }
+  
+  /**
+   * @return an iterator over the positive count values 
+   */
+  public Collection<Long> values(){
+    return counters.values();
+  }
+  
+  /**
+   * @return an iterator over the keys corresponding to positive counts only
+   */
+  public Collection<Long> keys(){
+    return counters.keySet();
+  }
+  
+  /**
+   * @param key, should not be null.
+   * @return the exact count for that key.
+   */
+  public Long get(Long key){
```

Comment:
Why does the user need the Long vs a long as the return value? especially if it could be null?  Again, hide the use of the object wrappers from the user.  This will also eliminate the need to check for nulls, as the run-time will automatically throw a null exception if the Long is null in attempting to convert to long.

Be aware that in Java, "assert" is enabled during unit tests, but by default disabled for normal operation.  The proper way to check parameters provided by the user is to check:  
if (key == null) throw new IllegalArgumentException("cause...");  
Thus the unit tests can and should exercise the limits of mathematical correctness, and then use asserts within the code to validate internal state.   Asserts can be enabled for a running program by adding "-ea" to the JVM arguments.   But in big pipeline operations this is only done as a last resort when troubleshooting a problem.